### ISSUE TITLE
Question on Wan2.2 Finetune method details

### ISSUE THREAD

**Issue by YixuannnL** [tdrussell/diffusion-pipe#357]
I'm new to MoE model training so i not so sure about the details.

In my understanding of this repo, the way this repo train MoE Wan2.2 model is: train low noise model and high noise model separately, on the condition of ensuring the timestep range of each model(0-0.85 for low noise, 0.85-1 for high noise).
Why not train both model together when finetuning? Does the Wan team do training not together? Why it is suitable to train separately like this repo do?

**Comment by kabachuha** [comment]
The Wan2.2 team is radio-silent on training details, however, they link to DiffSynth studio, where it fine-tunes Wan with two different script launches.
The batches of Wan images operate independently, basically the model is split in two fine-tunes of Wan2.1 turning on at different timesteps. It would be nice to know whether they employed some loss to keep the model representation together (semantically and graphically), like [RePA](https://arxiv.org/abs/2410.06940) between two models transformer layers at parallel batches, but we won't know it unfortunately, because they didn't even release the technical report this time. The other constraint is 2+ memory usage, if it's done in a single script.

If you want to simply parallelize training on 2GPUs, you can select the gpus needed in deepspeed/cuda devices and launch the high/low scripts in parallel (using tmux, for example)

**Comment by YixuannnL** [comment]
> The Wan2.2 team is radio-silent on training details, however, they link to DiffSynth studio, where it fine-tunes Wan with two different script launches. The batches of Wan images operate independently, basically the model is split in two fine-tunes of Wan2.1 turning on at different timesteps. It would be nice to know whether they employed some loss to keep the model representation together (semantically and graphically), like [RePA](https://arxiv.org/abs/2410.06940) between two models transformer layers at parallel batches, but we won't know it unfortunately, because they didn't even release the technical report this time. The other constraint is 2+ memory usage, if it's done in a single script.
> 
> If you want to simply parallelize training on 2GPUs, you can select the gpus needed in deepspeed/cuda devices and launch the high/low scripts in parallel (using tmux, for example)

Thank you so much for your reply!
You said, " you can select the gpus needed in deepspeed/cuda devices and launch the high/low scripts in parallel (using tmux, for example)." This means i launch them at the same time, but there is no interaction(loss func calc, etc.) between the two, right? ( Not sure if i understand your words correctly.)

**Comment by kabachuha** [comment]
Yes, the processes are completely isolated and run a different set of models, because they are basically two different programs.

I case of deepspeed, you should be a bit cautious, because the deepspeed launcher on the same file spawns multiple joined communicating processes. To isolate the processes, you can either set "CUDA_VISIBLE_DEVICES=[YOUR DEVICE NUMBER]" or add "--include" line to the launcher CLI command. This paragraph in deepspeed describes how to do this

https://www.deepspeed.ai/getting-started/#resource-configuration-single-node

**Comment by jakehemmerle** [comment]
to run two (or more) trainings, include  `--master_port 25001` as an argument for BOTH train.py and deepspeed. Specify the GPU with `--include localhost:0` for the GPU at 0, or `--include localhost:2,3` for the two at index 2 and 3.
